src/jshint.js

```diff
@@ -2598,7 +2669,7 @@ var JSHINT = (function() {
         isNecessary =
           (rbp > first.lbp) ||
           (rbp > 0 && rbp === first.lbp) ||
-          (!isEndOfExpr() && last.lbp < state.tokens.next.lbp);
+          (!isEndOfExpr() && last.rbp < state.tokens.next.lbp);
```

Comment:
Up until now, all the `rbp` and `lbp` were interchangeable. The exponentiation operator is the first feature to need distinct values, and that matters right here. This is what the somewhat lengthy comment in the `symbol` function is intended to explain.